### PR TITLE
[Build] fix travis.yml

### PR DESCRIPTION
**Pull request by crowning-**
References: https://github.com/dashpay/dash/pull/1129 and https://github.com/dashpay/dash/pull/1130

With this fix our builds over at travis-ci.org should work again.

~~Stole~~ Took some inspiration from several Bitcoin commits from the last couple of months, so a big "Thank you" to the Bitcoin developers.

(BTW, their builds have problems as well right now :-/ https://github.com/bitcoin/bitcoin/pull/9088#partial-pull-merging)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,78 +32,34 @@ env:
...
+  matrix:
+# ARM
+    - HOST=arm-linux-gnueabihf PPA="ppa:bitcoin/bitcoin" PACKAGES="g++-arm-linux-gnueabihf" DEP_OPTS="NO_QT=1" CHECK_DOC=1 GOAL="install" BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports"
+# Win32
+    - HOST=i686-w64-mingw32 DPKG_ADD_ARCH="i386" DEP_OPTS="NO_QT=1" PPA="ppa:bitcoin/bitcoin" PACKAGES="python3 nsis g++-mingw-w64-i686 wine1.6 bc" RUN_TESTS=true GOAL="install" BITCOIN_CONFIG="--enable-reduce-exports"
+# 32-bit + dash
+    - HOST=i686-pc-linux-gnu PPA="ppa:bitcoin/bitcoin" PACKAGES="g++-multilib bc python3-zmq" DEP_OPTS="NO_QT=1" RUN_TESTS=true GOAL="install" BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports LDFLAGS=-static-libstdc++" USE_SHELL="/bin/dash" PYZMQ=true
+# Win64
+    - HOST=x86_64-w64-mingw32 DPKG_ADD_ARCH="i386" DEP_OPTS="NO_QT=1" PPA="ppa:bitcoin/bitcoin" PACKAGES="python3 nsis g++-mingw-w64-x86-64 wine1.6 bc" RUN_TESTS=true GOAL="install" BITCOIN_CONFIG="--enable-reduce-exports"
+# bitcoind
```
Comment (UdjinM6) [inline .travis.yml]: dashd? ;)

File: .travis.yml [hunk 2]
```diff
@@ -24,78 +32,34 @@ env:
...
+    - HOST=x86_64-unknown-linux-gnu PPA="ppa:bitcoin/bitcoin" PACKAGES="bc python3-zmq" DEP_OPTS="NO_QT=1 NO_UPNP=1 DEBUG=1" RUN_TESTS=true GOAL="install" BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports CPPFLAGS=-DDEBUG_LOCKORDER" PYZMQ=true
```
Comment (UdjinM6) [inline .travis.yml]: missing `-DENABLE_DASH_DEBUG` flag

File: .travis.yml [hunk 3]
```diff
@@ -114,8 +78,7 @@ script:
     - ./configure --cache-file=../config.cache $BITCOIN_CONFIG_ALL $BITCOIN_CONFIG || ( cat config.log && false)
     - make $MAKEJOBS $GOAL || ( echo "Build failure. Verbose build follows." && make $GOAL V=1 ; false )
     - export LD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/depends/$HOST/lib
-    - if [ "$RUN_TESTS" = "true" -a "$WINE" != "true" ]; then make check; fi
-    - if [ "$RUN_TESTS" = "true" -a "$WINE" = "true" ]; then wine  src/test/test_dash.exe && wine src/qt/test/test_dash-qt.exe ; fi
+    - if [ "$RUN_TESTS" = "true" ]; then make $MAKEJOBS check VERBOSE=1; fi
```
Comment (UdjinM6) [inline .travis.yml]: Just to clarify - no more WINE option i.e. no more Windows tests? Is this intentional?

File: .travis.yml (hunk 1 above)
Comment (crowning-) [inline .travis.yml]: :-)
I'll try to add some descriptive names later anyway, so I can change this.

File: .travis.yml (hunk 3 above)
Comment (crowning-) [inline .travis.yml]: During my tests the wine test always failed, but that might have been side-effects of other problems.

I'll re-add them and see what happens.

**Comment by crowning-** [comment]
Update: right now I cannot verify whether my fix for testing with wine works or not, because Travis times out while build some the dependencies (which have worked before), so I guess Travis is just under a heavy load.
IMO you can merge this (all other tests run smoothly) and I'll re-check the wine tests for the Windows builds tomorrow or so...

**Comment by UdjinM6** [comment]
Looks good to me 👍  I'll try to restart windows build job a few times before merging it however to see if the issue persists.

**Comment by crowning-** [comment]
> I'll try to restart windows build job a few times

How do you do this without a new commit?

BTW, my first wine commit complained about the headless Travis installation (no $DISPLAY), so I deployed and configured a virtual display for wine.  Would be great if that works, however there are other options as well.

**Comment by UdjinM6** [comment]
> How do you do this without a new commit?

screen shot 2016-11-07 at 00 51 59

**Comment by UdjinM6** [comment]
Interesting... I restarted windows build jobs few times for both this (https://travis-ci.org/dashpay/dash/builds/173736623) and no DISPLAY (https://travis-ci.org/dashpay/dash/builds/173679891). The former seems to always fail, the later finishes in ~15-20 minutes, no issues. I guess it's not about heavy load, it's probably some internal travis issue. If that's true and there is no other way to enable win tests I think it's better to remove them/WINE flag/DISPLAY again for now (sorry :)) - always failing tests are meaningless anyway... Same as no tests of course but at least it will be easier to spot common errors via tests on other platforms. That's probably the reason bitcoin got them disabled, haven't check this assumption though.

**Comment by crowning-** [comment]
Interesting indeed. I'll remove the wine testing today after work so we can merge this PR, and after that I'll do more tests with my own Github repository (which is also linked to Travis) to get this resolved.

**Comment by crowning-** [comment]
On my repository all tests are green, here (with the same commit) the Win32 build gets stuck after 3 minutes or so (no error, just stuck), the rest runs fine. As if Travis had issues doing 2 identical builds...

Anyway, @UdjinM6 could you please re-trigger the Win32 build, and if it runs okay, merge?

**Pull request merged**